quixstreams/dataframe/dataframe.py

```diff
@@ -137,6 +138,7 @@ def __init__(
         processing_context: ProcessingContext,
         stream: Optional[Stream] = None,
         branches: Optional[Dict[str, Stream]] = None,
+        source: Optional[BaseSource] = None,
```

Comment:
Can we make the SDF class unaware of the sources? 
It would reduce the number of dependencies and simplify the code overall